src/control/lib/control/pool.go

```diff
@@ -429,6 +429,7 @@ type (
 	// PoolQueryResp contains the pool query response.
 	PoolQueryResp struct {
 		Status int32  `json:"status"`
+		State  string `json:"state"`
```

Comment:
I think that if you change the type to `system.PoolServiceState`, that type's stringer will automatically take care of the conversion to string for JSON output. As noted in another comment, it would probably be best to add a new `Degraded` state.

Reply:
Updated the code based on discussion, and it's been implemented